server/src/main/java/org/opensearch/common/lucene/search/function/FunctionScoreQuery.java

```diff
@@ -70,10 +71,27 @@
     public static class FilterScoreFunction extends ScoreFunction {
         public final Query filter;
         public final ScoreFunction function;
+        public final String queryName;
 
+        /**
+         * Creates a FilterScoreFunction with query and function.
+         * @param filter filter query
+         * @param function score function
+         */
         public FilterScoreFunction(Query filter, ScoreFunction function) {
+            this(filter, null, function);
+        }
+
+        /**
+         * Creates a FilterScoreFunction with query and function.
+         * @param filter filter query
+         * @param queryName filter query name
+         * @param function score function
+         */
+        public FilterScoreFunction(Query filter, @Nullable String queryName, ScoreFunction function) {
```

Comment:
@reta it's only a suggestion, maybe worth passing this brand new optional (nullable) argument to the end of the signature (and similar in other places)? wdyt? 🙂 
```suggestion
        public FilterScoreFunction(Query filter, ScoreFunction function, @Nullable String queryName) {
```
👉 in other constructors here in the file
👉 https://github.com/opensearch-project/OpenSearch/pull/2244/files#diff-9d4b4f755d23235ec5bd75b1a32ced38e6421271561522097f09c35fc1124112R89

Reply:
Should I wait for this to merge?

Reply:
Yes, please, already on it (was off today) :-)

Reply:
Done, thanks @dblock !